src/DataTypes/Serializations/SerializationEnum.h

```diff
@@ -1,20 +1,32 @@
 #pragma once
 
+#include <memory>
 #include <DataTypes/Serializations/SerializationNumber.h>
 #include <DataTypes/EnumValues.h>
+#include <DataTypes/DataTypeEnum.h>
 
 namespace DB
 {
 
 template <typename Type>
-class SerializationEnum : public SerializationNumber<Type>, public EnumValues<Type>
+class SerializationEnum : public SerializationNumber<Type>
 {
 public:
     using typename SerializationNumber<Type>::FieldType;
     using typename SerializationNumber<Type>::ColumnType;
-    using typename EnumValues<Type>::Values;
-
-    explicit SerializationEnum(const Values & values_) : EnumValues<Type>(values_) {}
+    using Values = EnumValues<Type>::Values;
+
+    SerializationEnum() = delete;
+    /// To explicitly create a SerializationEnum from Values
```

Comment:
Please replace l. 19-29 by

```cpp
/// SerializationEnum can be constructed in two ways:
/// - Make a copy of the Enum name-to-type mapping.
/// - Only store a reference to an existing mapping. This is faster if the Enum has a lot of different values or if SerializationEnum is
///   constructed very frequently. Make sure that the pointed-to mapping has a longer lifespan than SerializationEnum!

explicit SerializationEnum(const Values & values_)
    : own_enum_values(values_), ref_enum_values(&own_enum_values.value())
{
}

explicit SerializationEnum(const std::shared_ptr<const DataTypeEnum<Type>> & enum_type)
    : own_enum_type(enum_type), ref_enum_values(static_cast<const EnumValues<Type> *>(enum_type.get()))
{
}
```

(note that the default ctor, l. 19, is implicitly deleted when non-default ctors exist)